lnrpc/routerrpc/router_backend.go

```diff
@@ -864,6 +868,21 @@ func (r *RouterBackend) extractIntentFromSendRequest(
 	}
 	payIntent.FirstHopCustomRecords = firstHopRecords
 
+	// If the experimental endorsement signal is not already set, propagate
+	// a zero value field if configured to set this signal.
+	if r.SetExperimentalEndorsement() {
+		if payIntent.FirstHopCustomRecords == nil {
+			payIntent.FirstHopCustomRecords = make(
+				map[uint64][]byte,
+			)
+		}
+
+		t := uint64(lnwire.ExperimentalEndorsementType)
+		if _, set := payIntent.FirstHopCustomRecords[t]; !set {
+			payIntent.FirstHopCustomRecords[t] = []byte{0}
```

Comment:
I think it may be nice to lift the endorsement byte semantics out to a set of constants or some domain type?

Reply:
very nice cleanup 🥇